docs/reference/filebeat/file-identity.md

```diff
@@ -0,0 +1,189 @@
+---
+mapped_pages:
+  - https://www.elastic.co/guide/en/beats/filebeat/current/file-identity.html
+---
+
+# How to choose file identity for filestream [file-identity]
+
+## What is file identity?
+
+File identity is how [filestream input](/reference/filebeat/filebeat-input-filestream.md) differentiates one file from another in order to keep track of ingestion progress on each file.
+
+Files can be renamed, moved or replaced during ingestion and filestream must be able to handle these changes without data duplication or data loss. It's not as straightforward as it might seem.
+
+One of the most common examples is log rotation:
+
+* Let's say we just track files by their names / paths.
+* Imagine that after reaching 10 MB in size a log file `important.log` gets renamed to `important.log.old` and a new file is started in its place.
+* This log rotation happens before we reach the end of file (EOF) on `important.log`.
+
+Depending on your configuration 2 different things can happen:
+
+1. If you configured to exclude `*.old` files from ingestion you would have data loss.
+2. If you ingest from `*.old` - massive data duplication.
+
+Why would this happen? File names are not stable (non-changing) identifiers. This means we cannot use names to track the progress of ingestion on each file also known as offset. The offset on `important.log` was somewhere in the middle, let's say `offset: 145400` but since it got renamed to `important.log.old`, we detect this file as new and we start ingesting from the beginning - `offset: 0`. Therefore we re-ingest the same data again.
```

Comment:
```suggestion
Why would this happen? File names are not stable (non-changing) identifiers. This means that we cannot use names to track the progress of ingestion on each file also known as the offset. The offset on `important.log` was somewhere in the middle, for example `offset: 145400` but since the file was renamed to `important.log.old`, we detect this file as new and we start ingesting from the beginning, from `offset: 0`. Therefore we re-ingest the same data again.
```